ast/compile.go

```diff
@@ -284,6 +285,8 @@ func NewCompiler() *Compiler {
 		{"RewriteEquals", "compile_stage_rewrite_equals", c.rewriteEquals},
 		{"RewriteDynamicTerms", "compile_stage_rewrite_dynamic_terms", c.rewriteDynamicTerms},
 		{"CheckRecursion", "compile_stage_check_recursion", c.checkRecursion},
+		// must be run after CheckRecursion
```

Comment:
[nit] can we put that comment on the same line? right now, it could also apply all that follows 🤔

Reply:
This also applies to `compile_stage_check_types` below. But perhaps it's better to put this on _both_ lines ..